numpy/core/src/multiarray/datetime.c

```diff
@@ -330,6 +342,16 @@ convert_datetimestruct_to_datetime(PyArray_DatetimeMetaData *meta,
                       dts->ps / 1000;
                 break;
             case NPY_FR_ps:
+                /* only +- 106 days around epoch tolerated */
+                if (abs(get_datetimestruct_seconds(dts)) > 
+                        106 * 24 * 60 * 60) {
+                    /* reject with seconds resolution */
+                    PyErr_SetString(PyExc_ValueError,
```

Comment:
Not clear to me whether `ValueError` or `OverflowError` is the better choice. Would be good to put some justification for the choice in the commit message, whichever way you go.

Reply:
I was hoping I could replace `106 * 24 * 60 * 60` with a sufficiently high precision check using a formula with `NPY_INT64_MAX`, but if I understand correctly it could be too late by the time the calculation propagates down to seconds.